apps/balancer-tools/src/app/poolsimulator/(components)/SwapCurve.tsx

```diff
@@ -236,6 +257,32 @@ export function SwapCurve() {
     );
   }
 
+  if (initialData.poolType && initialData.poolParams?.beta) {
+    const betaLimits = getBetaLimitIndexes({
+      ...initialAmounts,
+      analysisTokenInitialBalance: analysisToken.balance,
+      tabTokenInitialBalance: currentTabToken.balance,
+      beta: initialData.poolParams.beta,
+    });
+
+    data.push(
+      createBetaRegionDataObject(betaLimits[0], betaLimits[1], "Initial"),
+    );
+  }
+
+  if (customData.poolType && customData.poolParams?.beta) {
+    const betaLimits = getBetaLimitIndexes({
+      ...customAmounts,
+      analysisTokenInitialBalance: analysisToken.balance,
+      tabTokenInitialBalance: currentTabToken.balance,
+      beta: customData.poolParams.beta,
+    });
+
+    data.push(
+      createBetaRegionDataObject(betaLimits[0], betaLimits[1], "Initial"),
```

Comment:
to keep it cleaner I'd have done something like `for stepData in [initialData, customData]:`